internal/providers/docker/components/container/lifecycle.go

```diff
@@ -42,6 +42,22 @@ func (c *Container) create(ctx context.Context) error {
 			StartPeriod: time.Duration(c.Spec.Healthcheck.StartPeriod),
 		}
 	}
+
+	cmd := strslice.StrSlice(c.Spec.Command)
+	if len(cmd) == 0 {
+		cmd = c.State.Image.Command
+	}
+
+	entrypoint := strslice.StrSlice(c.Spec.Entrypoint)
+	if len(cmd) == 0 {
+		entrypoint = c.State.Image.Entrypoint
+	}
+
+	workingDir := c.Spec.WorkingDir
+	if workingDir == "" {
+		workingDir = c.State.Image.WorkingDir
+	}
+
```

Comment:
super nit-picky, but here you are creating some locals and then assigning them in the `container.Config{` literal. However, right below that, we've got a bunch of code that is assigning directly in to the `containerCfg` object. The way you did it means that you can see a full list of assigned properties in one literal, but it also means having to name temporaries all over the place. Either way is fine, but it would be nice to follow the pattern.

```go
containerCfg.WorkingDir = c.Spec.WorkingDir
if containerCfg.WorkignDir == "" {
  containerCfg.WorkingDir = c.State.Image.WorkingDir
}
```